modules/core/lib/Auth/UserPassBase.php

```diff
@@ -169,6 +169,22 @@ public function isRememberMeChecked() {
 	public function authenticate(&$state) {
 		assert('is_array($state)');
 
+
+		if (isset($state['core:auth:username']) && isset($state['core:auth:password'])) {
+			/* It looks like we are doing HTTP Basic authentication. */
```

Comment:
Another comment on coding style. Multi-line comment delimiters should be used for comments that are truly multi-line. One-line comments like this should use their corresponding syntax `//`.